doc/arrays.md

```diff
@@ -0,0 +1,107 @@
+# Modifying arrays
+Technically, YAML allows you to create arrays of everything that is supported as a type. However, when dealing with BOSH deployment manifests, you typically encounter either
+- an array of simple types such as strings or numbers,
+- or an array of maps.
+An array of maps follows a common schema: One of the map entries serves as an identifier. This identifier gives the array entry a name and enables it to be referenced it internally.
+
+Simple array:
+```yml
+list:
+- 192.168.1.1
+- 192.168.1.2
+```
+
+Array of maps:
+```yml
+list:
+- name: Controller
+  version: v1
+  active: true
+- name: Logger
+  version: v4
+  active: false
+```
+
+In this example, the map
+```yml
+name: Controller
+version: v1
+active: true
+```
+is an array entry.
+
+Spruce provides operators to modify a given array:
+- `(( append ))`
+- `(( prepend ))`
+- `(( insert ... ))`
+- `(( delete ... ))`
+- `(( replace ))`
+- `(( merge ))`, and `(( merge ... ))` respectively
+
+Some of these operators can be used multiple times and/or in combination with others. These operators are: `(( append ))`, `(( prepend ))`, `(( insert ... ))`, and `(( delete ... ))`. With the exception of `(( delete ... ))`, the respective operator applies to all following array entries until the end of the array or a new operator. Obviously, the delete operator always stands alone. Array entries that cannot be attached to a preceding operator are called _orphaned_ entries and will result in an error.
+```yml
+list:
+- (( append ))
+- "The new first entry"
+- "The new second entry"
+
+- (( prepend ))
+- "The new second to last entry"
+- "The new last entry"
+```
+
+To emphasize the new annotation an empty line was added. This is technically not needed, however it increases the readability.
+
+## Operators that work the same for either type of array
+- `(( append ))` - The append operator tells Spruce to place the following content before the first existing entry.
```

Comment:
I believe the description for append + prepend are swapped. prepend goes in front of the first entry, append goes after?